test/examples_test.go

```diff
@@ -0,0 +1,399 @@
+		},
+		"../docs/user-guide/node-selection": {
+			"pod": &api.Pod{},
+			"pod-with-node-affinity": &api.Pod{},
+		},
+		"../docs/admin/resourcequota": {
+			"best-effort":       &api.ResourceQuota{},
+			"compute-resources": &api.ResourceQuota{},
+			"limits":            &api.LimitRange{},
+			"namespace":         &api.Namespace{},
+			"not-best-effort":   &api.ResourceQuota{},
+			"object-counts":     &api.ResourceQuota{},
+		},
+		"../docs/user-guide/secrets": {
+			"secret-pod":     &api.Pod{},
+			"secret":         &api.Secret{},
+			"secret-env-pod": &api.Pod{},
+		},
+	}
+
+	capabilities.SetForTests(capabilities.Capabilities{
+		AllowPrivileged: true,
+	})
+
+	for path, expected := range cases {
+		tested := 0
+		err := walkJSONFiles(path, func(name, path string, data []byte) {
+			expectedType, found := expected[name]
+			if !found {
+				t.Errorf("%s: %s does not have a test case defined", path, name)
+				return
+			}
+			tested++
+			if expectedType == nil {
+				t.Logf("skipping : %s/%s\n", path, name)
+				return
+			}
+			if strings.Contains(name, "scheduler-policy-config") {
+				if err := runtime.DecodeInto(schedulerapilatest.Codec, data, expectedType); err != nil {
+					t.Errorf("%s did not decode correctly: %v\n%s", path, err, string(data))
+					return
+				}
+				//TODO: Add validate method for &schedulerapi.Policy
+			} else {
+				codec, err := testapi.GetCodecForObject(expectedType)
+				if err != nil {
+					t.Errorf("Could not get codec for %s: %s", expectedType, err)
+				}
+				if err := runtime.DecodeInto(codec, data, expectedType); err != nil {
+					t.Errorf("%s did not decode correctly: %v\n%s", path, err, string(data))
+					return
+				}
+				if errors := validateObject(expectedType); len(errors) > 0 {
+					t.Errorf("%s did not validate correctly: %v", path, errors)
+				}
+			}
+		})
+		if err != nil {
+			t.Errorf("Expected no error, Got %v on Path %v", err, path)
+		}
+		if tested != len(expected) {
+			t.Errorf("Directory %v: Expected %d examples, Got %d", path, len(expected), tested)
+		}
+	}
+}
+
+// This regex is tricky, but it works.  For future me, here is the decode:
+//
+// Flags: (?ms) = multiline match, allow . to match \n
+// 1) Look for a line that starts with ``` (a markdown code block)
+// 2) (?: ... ) = non-capturing group
+// 3) (P<name>) = capture group as "name"
+// 4) Look for #1 followed by either:
+// 4a)    "yaml" followed by any word-characters followed by a newline (e.g. ```yamlfoo\n)
+// 4b)    "any word-characters followed by a newline (e.g. ```json\n)
+// 5) Look for either:
+// 5a)    #4a followed by one or more characters (non-greedy)
+// 5b)    #4b followed by { followed by one or more characters (non-greedy) followed by }
+// 6) Look for #5 followed by a newline followed by ``` (end of the code block)
+//
+// This could probably be simplified, but is already too delicate.  Before any
+// real changes, we should have a testscase that just tests this regex.
```

Comment:
Do we have many documents with examples embedded in markdown, rather than included separately? If the number isn't too high, we could probably get rid of the regex and walking over md files.